### PR TITLE
feat: onchain payment

### PR DESCRIPTION
**Pull request by huynguyenh**
**What does this PR do?**

-   [x] My bugfix

**How to test**

-   Create a detailed description of what you need to do to set this PR up. ie: Does it need migrations? Do you need to install something?
-   Create a step by step list of what the engineer needs to do to test.

**Edge cases**

-   What are the edge cases that you might see in this PR?

**Media (Loom or gif)**

-   Insert media here (if applicable)

### CODE REVIEW

File: cmd/watch-tx/watch_tx.go
Comment (huynguyenh) [inline cmd/watch-tx/watch_tx.go:1]: move to appmain

File: internal/app/api/handler/payrequest/pay_request.go [hunk 1]
```diff
@@ -49,26 +49,53 @@ func (h *handler) GenerateCode(c *gin.Context) {
...
-		logger.Error(err, "[handler.Claim] controller.PayRequest.ClaimPayment() failed")
+		logger.Error(err, "[handler.ClaimToMochiWallet] controller.PayRequest.ClaimOffChainPayment() failed")
+		c.JSON(err.StatusCode(), view.ToErrorResponse(err))
+		return
+	}
+
+	c.JSON(http.StatusOK, view.ToRespSuccess("Claimed successfully"))
+}
+
+func (h *handler) ClaimToPublicKey(c *gin.Context) {
```
Comment (huynguyenh) [inline internal/app/api/handler/payrequest/pay_request.go:79]: ClaimToOnchainAddress

File: internal/app/api/handler/payrequest/pay_request.go [hunk 2]
```diff
@@ -49,26 +49,53 @@ func (h *handler) GenerateCode(c *gin.Context) {
...
+	logger.Debug("api call ", c.Request.RequestURI)
+	defer logger.Debug("api finish ", c.Request.RequestURI)
+
+	uri := dto.ClaimPaymentUriRequest{}
```
Comment (huynguyenh) [inline internal/app/api/handler/payrequest/pay_request.go:83]: dto.ClaimPaymentUriRequest -> dto.ClaimPaymentRequest ?

File: internal/app/checkexpiredpayment/run.go [hunk 3]
```diff
@@ -44,9 +45,9 @@ func (c *checkExpiredPayment) Run() {
 		return
 	}
 	for _, pr := range expired {
-		err = c.ctrl.HandleExpiredCode(&pr)
+		err = c.ctrl.HandleFailedPayment(&pr, constant.PayRequestExpired)
```
Comment (huynguyenh) [inline internal/app/checkexpiredpayment/run.go:48]: sao chỗ này đang từ expire r thành failed luôn rồi
